asr-python-advanced-finetune-am-noise-augmentation.ipynb

```diff
@@ -0,0 +1,541 @@
+    "\n",
+    "test_manifest = data_dir + '/an4/test_manifest.json'\n",
+    "\n",
+    "run = f\"python NeMo/scripts/dataset_processing/add_noise.py \\\n",
+    "    --input_manifest={train_manifest} \\\n",
+    "    --noise_manifest={train_noise_manifest} \\\n",
+    "    --snrs 0 5 10 15 \\\n",
+    "    --out_dir={final_data_dir}\"\n",
+    "\n",
+    "! $run"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above script generates a `.json` manifest file each for every SNR value, i.e., one manifest file each for 0, 5, 10 and 15db SNR.  \n",
+    "Let's combine all the manifests into a single manifest for training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = f\"cat {final_data_dir}/manifests/train* >{final_data_dir}/manifests/noisy_train.json\"\n",
+    "!{run}\n",
+    "\n",
+    "print(\"Noise-augmented training dataset created at\", final_data_dir + \"/manifests/noisy_train.json\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Test dataset\n",
+    "\n",
+    "Let's create a noise-augmented evaluation dataset using the an4 test dataset, by adding noise at 5 dB, using a NeMo script"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Data augmention - Add noise to test set.\n",
+    "run = f\"python NeMo/scripts/dataset_processing/add_noise.py \\\n",
+    "    --input_manifest={test_manifest} \\\n",
+    "    --noise_manifest={test_noise_manifest} \\\n",
+    "    --snrs=5 \\\n",
+    "    --out_dir={final_data_dir}\"\n",
+    "\n",
+    "# For some reason, adding --user=$(id -u):$(id -g) breaks the script\n",
+    "!$run\n",
+    "\n",
+    "print(\"Noise-augmented testing dataset created at\", final_data_dir+\"/test_manifest\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Noise-augmented training manifest and data are created at `{working_dir}/noise_data/noisy_train.json` and `{working_dir}/noise_data/train_manifest` respectively.**  \n",
+    "**Noise-augmented testing manifest and data are created at `{working_dir}/noise_data/manifests/test_manifest_test_noise_5db.json` and `{working_dir}/noise_data/test_manifest` respectively.**  \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### We don't need the NeMo container anymore, so let's get rid of it"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "###  Fine-tuning the ASR model\n",
+    "To finetune the ASR model with the augmented datasets that we just created, you can \n",
+    "proceed to [this tutorial](https://github.com/nvidia-riva/tutorials/blob/stable/asr-python-advanced-finetune-am-conformer-ctc-tao-finetuning.ipynb)."
```

Comment:
This path needs to be changed to the path of the other finetuning tutorial after it is merged.